src/OpenTelemetry.Abstractions/Trace/IDecision.cs

```diff
@@ -0,0 +1,34 @@
+﻿// <copyright file="IDecision.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Trace
+{
+    using System.Collections.Generic;
+
+    public interface IDecision
```

Comment:
This implies virtual call to get `IsSampled` on all spans. Can we avoid it? 

we also need to have a discussion about interfaces and layers of abstraction. So I hope (and plan) we'll remove all interfaces - they are not extendable and switch them to abstract base classes where we need an abstraction.

I wonder if we can have 
- `class Decision` that contains implementation as well 
- with `IsSampled` non-virtual property
- protected ctor taking bool
- two public static properties `SampledIn`, `SampleOut` (whatever).

 If someone will ever want to implement their own decision - they are free to inherit from this class and extend it (but non likely will ever do this)

Reply:
If we doing this, I'll make Decision a struct than =) When there is no dictionary - it must work even faster.

Reply:
And we will need public ctor that will take bool. So some fancy samplers may create one

Reply:
Cool! The only problems with structs is that it's not possible to inherit from it, so if someone will want to extend decision, they won't be able to.

I don't know why and if we'd want abstraction and fine with struct. But if default sampling decisions are preallocated, there is no concern with class

Reply:
I can't really think of a reason that someone would extend Decision, so I think struct would be fine here?